src/compiler/ruby_generator.cc

```diff
@@ -37,16 +38,61 @@ using std::vector;
 namespace grpc_ruby_generator {
 namespace {
 
+// returns the full canonical message name
+grpc::string GetCanonicalMessageType(const Descriptor* msg) {
+  const Descriptor* top_level_msg;
+  const FileDescriptor* file_containing_msg;
+  grpc::string msg_full_name;
+  grpc::string msg_name;
+  grpc::string msg_proto_pkg;
+  grpc::string resolved_namespace;
+  grpc::string res(msg->full_name());
```

Comment:
nit: can we move declaration/initialization of `res` down to just above where it's used?